src/lib/drivers/driver.js

```diff
@@ -86,6 +86,28 @@ class DriverBase {
     return Promise.reject(new Error('Not implemented'));
   }
 
+  evaluateAsync(asyncExpression) {
+    return new Promise((resolve, reject) => {
+      // Inject the call to capture inspection.
+      const expression = `window.__inspect = inspect;${asyncExpression}`;
+
+      this.on('Runtime.inspectRequested', value => {
+        // Tidy up the injected call.
+        this.sendCommand('Runtime.evaluate', {
+          expression: 'delete window.__inspect'
+        }).then(_ => resolve(value));
```

Comment:
Right, we're currently in series. I agree that if we parallelize we'd do this, but if anything I imagine we'd parallelize the audit phase, not this.